src/lib/up-storage.ts

```diff
@@ -329,6 +330,12 @@ class ProxyStorage implements IProxy {
         this.logger.error(ERROR_CODE.token_required);
         this._throwErrorAuth(ERROR_CODE.token_required);
       }
+    } else if (_.isNil(tokenConf.file) === false) {
+      try {
+        token = fs.readFileSync(tokenConf.file, { encoding: 'utf8' });
```

Comment:
Read a sync file for every request blocks the event loop for incoming requests I don´t think this is very performant.

Reply:
You are right, i think i could cache the file contents with a specified TTL. However retrieval of a few bytes for that file is served directly from FS cache i didn't notice any performance overhead.

You are right, i will reformulate the proposal with possibility to cache the contents with a TTL ?

Reply:
Have you just read the file and add it to the env variables before even start verdaccio? I think you can solve it without touch the core. Verdaccio can be executed programmatically anyway.

Reply:
Hello, yes of course it works but once token is expired how would you update it ?

Reply:
imho like any other service, has to be restarted.

Reply:
That's exactly the point, let's say the token expires every hour (we have no control over this) then if we restart Verdaccio service while CI pipelines are using it, then then client jobs will fail. Therefore, my proposal was to enable dynamic update of the token without restart.

For example, in Logstash there is a mechanism to hot reload configuration through a SIGHUP signal.

In this case, maybe I could cache the token in memory and not reread it every request.

Another option would be to make a special route to update the uplink token.

I appreciate your feedbacks, thank you

Reply:
For the dotenv to reload variables, it is only half way since systemd env vars are immutable and cant be updated externally...

Reply:
Well, I have other ideas, create a middleware plugin that reads that file on every request (but you will pay the performance penalty, but also you can use lru-cache to mitigate it) and modify the env variables https://nodejs.org/api/process.html#processenv since is doable to do. It will be under your full control no core changes need it.

Reply:
I didn't know about plugins I will investigate this way it is a great idea, thanks you very much !